src/core/raster/qgssinglebandpseudocolorrenderer.cpp

```diff
@@ -84,6 +88,193 @@ void QgsSingleBandPseudoColorRenderer::setShader( QgsRasterShader* shader )
   mShader = shader;
 }
 
+void QgsSingleBandPseudoColorRenderer::createShader( QgsColorRamp* colorRamp, QgsColorRampShader::ColorRamp_TYPE colorRampType, bool clip )
```

Comment:
@wonder-sk , and where would the [Classification]Mode enum go? I don't mind moving it  (I trust your big-picture vision more than my own 😉), but the reason why I put it in the renderer to begin with is the implicit link between a number of parameters within that function and the renderer's settings.